### PR TITLE
fix: normalising ChatGenerators output

### PR DESCRIPTION
**Pull request by davidsbatista**
### Related Issues

- fixes #7687

### Proposed Changes:

- Something change in AmazonBedrock and the PR I merged had no effect

### How did you test it?

unit tests, integration tests, manual verification, instructions for manual tests

#### Test coverage

- `generators/amazon_bedrock/chat/adapters.py`is at 59%
- `generators/amazon_bedrock/chat/chat_generator.py` is at 51%

#### Some integrations tests are still being skipped, see the CI logs for details

```
cmd [1] | coverage run -m pytest -m integration --reruns 3 --reruns-delay 30 -x
============================= test session starts ==============================
platform linux -- Python 3.9.19, pytest-8.3.2, pluggy-1.5.0
rootdir: /home/runner/work/haystack-core-integrations/haystack-core-integrations/integrations/amazon_bedrock
configfile: pyproject.toml
plugins: rerunfailures-14.0, anyio-4.4.0
collected 152 items / 140 deselected / 12 selected
tests/test_chat_generator.py::TestMistralAdapter::test_default_inference_params[mistral.mistral-7b-instruct-v0:2] SKIPPED [  8%]
tests/test_chat_generator.py::TestMistralAdapter::test_default_inference_params[mistral.mixtral-8x7b-instruct-v0:1] SKIPPED [ 16%]
tests/test_chat_generator.py::TestMistralAdapter::test_default_inference_params[mistral.mistral-large-2402-v1:0] SKIPPED [ 25%]
```

#### Several warnings about parameters being ignored

```
WARNING  haystack_integrations.components.generators.amazon_bedrock.chat.adapters:adapters.py:76 Parameter 'top_k' is not allowed and will be ignored.
WARNING  haystack_integrations.components.generators.amazon_bedrock.chat.adapters:adapters.py:76 Parameter 'stop_sequences' is not allowed and will be ignored.
WARNING  haystack_integrations.components.generators.amazon_bedrock.chat.adapters:adapters.py:76 Parameter 'stream' is not allowed and will be ignored.

```

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py [hunk 1]
```diff
@@ -205,9 +205,21 @@ def run(
...
+                        response.meta["usage"]["completion_tokens"] = response.meta["usage"].pop("output_tokens")
+                elif "usage" not in response.meta:
+                    if "prompt_token_count" in response.meta:
+                        response.meta["usage"] = {}
+                        response.meta["usage"]["prompt_tokens"] = response.meta.pop("prompt_token_count")
+                    if "generation_token_count" in response.meta:
+                        response.meta["usage"]["completion_tokens"] = response.meta.pop("generation_token_count")
+                else:
+                    msg = "The meta key is not in the expected format."
+                    raise ValueError(msg)
```
[the remark this answers is not in the document]
Comment (anakin87) [reply integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:222]: I would not raise an exception. Usage is nice to have, but not essential.

[the remark this answers is not in the document]
Comment (davidsbatista) [reply integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:222]: I just wanted a way to make sure we catch all possible outputs from all possible models and normalising them.

**Comment by davidsbatista** [comment]
@vblagoje @anakin87 regarding the issues I raised in the PR comment - I will open an issue for it, you agree?

**Comment by anakin87** [comment]
On test coverage, etc.: see #801

In short, we split unit and integration tests to make integration tests run only if authenticated (not from forks).
- This causes the coverage numbers to be strange and probably incorrect.
- Selected tests are different depending on the unit/integration workflow step.

**Comment by davidsbatista** [comment]
ok I see the rational do to that, but since I couldn't successfully authenticate locally to test everything, I'm not sure something will fail when the full tests run on `main`

File: integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py [hunk 2]
```diff
@@ -208,9 +208,18 @@ def run(
...
-                response.meta["usage"]["completion_tokens"] = response.meta["usage"].pop("output_tokens")
+            if response.meta:
+                if "usage" in response.meta:
+                    if "input_tokens" in response.meta["usage"]:
+                        response.meta["usage"]["prompt_tokens"] = response.meta["usage"].pop("input_tokens")
+                    if "output_tokens" in response.meta["usage"]:
+                        response.meta["usage"]["completion_tokens"] = response.meta["usage"].pop("output_tokens")
+                else:
+                    if "prompt_token_count" in response.meta:
+                        response.meta["usage"] = {}
```
Comment (anakin87) [inline integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py:219]:
```suggestion
                    response.meta["usage"] = {}
                    if "prompt_token_count" in response.meta:
```

maybe it is safer to do that

**Review by anakin87 — COMMENTED** [review]
There is some issue with my suggestion.
Could you fix it?

**Comment by davidsbatista** [comment]
@anakin87 fixed

**Comment by anakin87** [comment]
This test is wrong
https://github.com/deepset-ai/haystack-core-integrations/blob/55c65af5ac338f0678537125d6fd0e92dfa689b6/integrations/amazon_bedrock/tests/test_chat_generator.py#L231

`meta` should be a dict: https://github.com/deepset-ai/haystack/blob/35b1215b00b88b2654ea50de2453e5f852e72328/haystack/dataclasses/chat_message.py#L33

I think you can fix the test and add back the previous suggestion.

**Review by anakin87 — APPROVED** [review]
looks good to me

**Pull request merged**